### PR TITLE
Remove optimizer unless code

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/PlanFragmentTest.java [hunk 1]
```diff
@@ -4032,47 +4032,6 @@ public void testMergeAggregateFailed() throws Exception {
                 "  0:OlapScanNode"));
     }
 
-    // todo(ywb) disable replicate join temporarily
-    public void testReplicatedJoin() throws Exception {
```
Comment (kangkaisen) [inline fe/fe-core/src/test/java/com/starrocks/sql/plan/PlanFragmentTest.java:4036]: why directory remove  this test?

Comment (Seaven) [reply fe/fe-core/src/test/java/com/starrocks/sql/plan/PlanFragmentTest.java:4036]: We need support replicate join later？

**Comment by Seaven** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/PlanFragmentTest.java (hunk 1 above)
Comment (kangkaisen) [reply fe/fe-core/src/test/java/com/starrocks/sql/plan/PlanFragmentTest.java:4036]: maybe.  I think we could keep the test code.

**Review by Youngwb — DISMISSED** [review]

**Review by kangkaisen — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**